datadog-monitors-downtime-handler/src/datadog_monitors_downtime/handlers.py

```diff
@@ -0,0 +1,199 @@
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL,
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = downtimes_api.DowntimesApi(api_client)
+        try:
+            api_instance.update_downtime(model.Id, downtime_body)
+        except ApiException as e:
+            LOG.error("Exception when calling DowntimeApi->update_downtime: %s\n" % e)
+            return ProgressEvent(status=OperationStatus.FAILED, resourceModel=model, message=e.body)
+
+    return read_handler(session, request, callback_context)
+
+
+@resource.handler(Action.DELETE)
+def delete_handler(
+    session: Optional[SessionProxy],
+    request: ResourceHandlerRequest,
+    callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS,
+        resourceModel=model,
+    )
+    LOG.info(f"Starting the {TYPE_NAME} Delete Handler")
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL,
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = downtimes_api.DowntimesApi(api_client)
+        try:
+            api_instance.cancel_downtime(model.Id)
+        except ApiException as e:
+            LOG.error("Exception when calling DowntimeApi->cancel_downtime: %s\n" % e)
+            return ProgressEvent(status=OperationStatus.FAILED, resourceModel=model, message=e.body)
+
+    return read_handler(session, request, callback_context)
+
+
+@resource.handler(Action.READ)
+def read_handler(
+    session: Optional[SessionProxy],
+    request: ResourceHandlerRequest,
+    callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+    LOG.info(f"Starting the {TYPE_NAME} Read Handler")
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL,
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = downtimes_api.DowntimesApi(api_client)
+        try:
+            api_resp = api_instance.get_downtime(model.Id)
+        except ApiException as e:
+            LOG.error("Exception when calling DowntimeApi->get_downtime: %s\n" % e)
+            return ProgressEvent(status=OperationStatus.FAILED, resourceModel=model, message=e.body)
+
+    LOG.info(f"Success retrieving downtime {api_resp}")
+
+    # Not Nullable attributes
+    model.Message = api_resp.message
+    model.MonitorTags = api_resp.monitor_tags
+    model.Scope = api_resp.scope
+    model.Timezone = api_resp.timezone
+    model.Start = api_resp.start
```

Comment:
We probably need to check those attributes are present in the response no ? Since they are not nullable, it's possible they won't be defined at all and this will raise ?